src/Symfony/Component/ExpressionLanguage/Node/GetAttrNode.php

```diff
@@ -97,10 +103,14 @@ public function evaluate(array $functions, array $values)
 
             case self::ARRAY_CALL:
                 $array = $this->nodes['node']->evaluate($functions, $values);
-                if (!\is_array($array) && !$array instanceof \ArrayAccess) {
+                if (!\is_array($array) && !$array instanceof \ArrayAccess && !(null === $array && $this->isNullCoalesce)) {
                     throw new \RuntimeException(sprintf('Unable to get an item of non-array "%s".', $this->nodes['node']->dump()));
                 }
 
+                if ($this->isNullCoalesce) {
+                    return $array[$this->nodes['attribute']->evaluate($functions, $values)] ?? null;
```

Comment:
This is going to evaluate the key even if $array is null, isn't it?

Reply:
Technically, YES!
The expression used to evaluate the key is going to be fired. However, I verified with the native syntax that PHP also evaluate the key even when `$array` is null.

For example dumping this expression in native PHP:
`null['key'] ?? 'default';`
will output 'default'.

When the key is retrieved using a dynamic syntax:
```
function evalKey() {return 'key';}
null[evalKey()] ?? 'default';
```
same output will be dumped.

Any exception thrown during key evaluation, will propagate up to the main evaluation level:
 ```
function evalKey() {throw new Exception();}
null[evalKey()] ?? 'default';
```
will throw an Exception.
That's why we should **allow** the key evaluation when `$array` is null to imitate this native behavior.

Reply:
That is unexpected! Can you please open an issue on php-src about this? I'd be happy to see what others think about this behavior there!

Reply:
Done!
=> [Issue #8520](https://github.com/php/php-src/issues/8520)
Let's see what others think..

Reply:
Great thanks. Can you please add a test case that ensures tests will break if the behavior changes, either when compiling or when evaluating?